### ISSUE TITLE
[Wii] 1.0.0.3 unexpected GUI behaviour / glitching at 240p

### ISSUE THREAD

**Issue by 240p4eva** [libretro/RetroArch#1463]
Using retroarch-wii-1.0.0.x.nightly-20141126.zip.  Wii is set to 480i, and outputting component to a Sony PVM-14L2.

1.Any 240p resolution that doesn't begin with 640x results in a garbled GUI, but the game underneath displays at the correct frequency.  For instance, using the PCE Mednafen core set to 512x232 results in a distorted and unreadable GUI (game is fine), while on 1.0.0.2 it was "pixel perfect".
2. When setting the video resolution, the resolutions no longer switch on the fly but require a button press to enact the change.  Is this intended behaviour?  It was much easier to spot compatible resolutions on my CRT when they changed as soon as selected...

**Comment by twinaphex** [comment]
> 2. When setting the video resolution, the resolutions no longer switch on the fly but require a button press to enact the change. Is this intended behaviour? It was much easier to spot compatible resolutions on my CRT when they changed as soon as selected...

Yes, that is intended behavior. I'm not going to have totally separate resolution code for every platform. It will all have to work uniform (i.e. this way) or else the code will have to be removed entirely because I'm not going to be adding even more stuff to my maintenance backlog for stupid stuff that should really be unified anyway.

**Comment by 240p4eva** [comment]
Thanks for clarifying--it's not like it's a big deal.  Didn't realize it was a nighmare behind the scenes the other way 'round.

Can you comment on the glitching?  Is it a known or repeatable phenomenon?  Would it help if I made a video of what is happening?

Thanks--and thanks for all the work so far!  You've breathed new life into my Wii...

**Comment by twinaphex** [comment]
> 1.Any 240p resolution that doesn't begin with 640x results in a garbled GUI, but the game underneath displays at the correct frequency. For instance, using the PCE Mednafen core set to 512x232 results in a distorted and unreadable GUI (game is fine), while on 1.0.0.2 it was "pixel perfect".

Not sure. It could be a regression as a result of some stuff @ToadKing changed earlier to do with the framebuffer. Hopefully it's not that though.

Really, we are short on manpower and we NEED more developers (and more developers active as well). It is becoming very stress-inducing and the workload keeps becoming bigger, bigger and bigger

**Comment by twinaphex** [comment]
Also, I just noticed you are testing an old build (1.0.0.3). Please, don't.

Recompile from source and check out the latest version. Please don't report things to us about some old build, all this does is cause more confusion and cause the release date to slip even further because we have to look into ancient bugs on some ancient version that is not reflective of the current version anyway.

**Comment by 240p4eva** [comment]
My apologies.  It was my (mis)understanding that the file I used was the newest "nightly" that had been produced for Wii.  Didn't realize I had to compile my own to actually get the "nightly", though that makes sense when you think about it.  Will look into how to compile and report back.

If the issue persists, would you rather I edit this thread's title and continue here, or close this ticket and open a new one?

**Comment by iKarith** [comment]
When I have libretro-super into a bit more buildbot-friendly shape, it shouldn't be hard to add Wii builds.  As I now have one to test them, I am somewhat interested.  It's on my todo list.

**Comment by 240p4eva** [comment]
@iKarith Sounds phenomenal!  I've never compiled anything in my life, and I'd be lying if I said it was going well.  I've got the pre-packaged toolchain, headers and libs in place, but I'm stuck at "A pre-existing libretro library needs to be present in the root directory in order to link RetroArch Wii. This file needs to be called libretro_wii.a."

This isn't a thread to cover what I assume are really the basics of compiling, but a pre-compiled nightly for the Wii would be a god-send.  I've signed up at the libretro forums and confirmed my account yet I have no posting privileges.  Sent a contact form support request explaining the situation but no response yet.

At this rate I'm not terribly optimistic I'll be able to compile this myself to help test it...

**Comment by iKarith** [comment]
Let me explain that: On most systems, including any other you might have used RetroArch on, RA is a program that loads multiple cores at runtime.  These DSO (dynamic shared object) files would be named something.so on most UNIXes, .dll on Windows, .dylib on a Mac, etc.

But the Wii (and the PSP actually) don't have enough RAM for that to work well, so we trade hogging drive/SD card space for saving RAM.  When you compile a core, say nestopia for example, it'll generate some .a file for the Wii.  You need to take the .a file from compiling the core and stick it into your RetroArch directory with the name libretro_wii.a before linking it (which is the last step of a compile…)

Basically, instead of one RetroArch program and a whole bunch of cores, each core gets its own copy of RetroArch compiled in.

Twinaphex says that this setup can actually be transparent, even letting you switch cores within RA.  I wonder though if the upcoming MIT libretro player (a simplified RA that works for one core only) might gain us some benefit on systems like the Wii.  Of course, you'd "switch" cores by running something different from the Homebrew Channel if we did that, but effectively that's already how it's done—you just don't see it happening.

**Comment by ToadKing** [comment]
I do not experience the 1st issue, and the 2nd one is by design.

**Comment by 240p4eva** [comment]
@iKarith Thanks for the background--that's quite enlightening and a rather ingenious way to work around the Wii's limitations.  I've succeeded in building a PC nestopia core (dll), but my wii builds fail.  Clearly I am not understanding something here...

@ToadKing Thanks for confirming--this re-invigorates my desire to figure out the compiling on my end!

**Comment by iKarith** [comment]
I'll have a look at the process in the next few days and see what I can do with it.

Joseph
Sent via mobile

**Comment by 240p4eva** [comment]
@iKarith Much appreciated.  A buildbot--*any* buildbot would do a better job than I'm doing at the moment.  Dear god...

**Comment by fr500** [comment]
btw, we have wii builds already, I just need to add more cores

**Comment by 240p4eva** [comment]
http://buildbot.libretro.com/nightly/wii/
Is this folder brand new?  I swear it wasn't there like...2 hours ago.  Will test it ASAP!

**Comment by 240p4eva** [comment]
That nightly...and again--excuse my ignorance, should it be bootable *as is* without further modification?  Because it boots me out back to the HBC after a few seconds, with or without a GC controller connected...

**Comment by fr500** [comment]
Yes it was not there.
Erase your older configs

**Comment by 240p4eva** [comment]
At first I only overwrote the older files with that nighly and RetroArch froze on boot twice.  Then I nuked the whole retroarch-wii folder in apps and copied over only the new files from the nightly.  This is when it failed to boot at all.  The launch screen in HBC read 1.0.0.2, if that matters...

**Comment by 240p4eva** [comment]
my Wii is a freshly modded (via ModMii) 4.3U, model RVL-001.  Last official RetroArch build worked.

**Comment by fr500** [comment]
no idea mate, I just tried it in dolphin I don't have a Wii, I'll ask @ToadKing but @twinaphex got it working earlier today

**Comment by 240p4eva** [comment]
Thanks for looking in to it.  I'm hoping to not bring horribly ignorant questions to the table, but I'm finding the whole RetroArch experience to be a bit of a maze from a layman's perspective.

If the contents of that nightly are supposed to run on a Wii as is...they don't on my end.  Either I need further direction in the treatment of nightlies, or it doesn't work.  Do you know if @twinaphex tested it in 480i over component?

I minute I get posting privileges over here
http://libretro.com/forums/showthread.php?t=2234&p=19829#post19829
I'll be sure to share my results there too.  My username is the same, 240p4eva.

**Comment by 240p4eva** [comment]
Answering my own question--the meta.xml lists 1.0.0.2, so that's expected behaviour for this nightly.

**Comment by 240p4eva** [comment]
@fr500 Apologies.  Just now saw where RetroArch houses the cfg files, and they hadn't been wiped.  About to test.

**Comment by fr500** [comment]
it's working alright here

**Comment by 240p4eva** [comment]
Working.  Where can I find a list of known issues for Wii, so that I don't waste time reporting things you guys know about?  

I.E. Core resolution settings don't stick after saving a new config.  FCEUMM running at 512x240 and saved, reloads at 640x480.  Or is this again intended behaviour?  Any way to override what I'm guessing is a "safety feature" for those who may select it on incompatible displays?

**Comment by 240p4eva** [comment]
I've been activated over on the forums and will take any concerns over there with the new nightlies.  The issue I reported does not appear to exist now.  I'm closing this ticket--thanks!

**Comment by iKarith** [comment]
I won't deny that it is a bit of a maze for the moment.  One of RA's 
founders (and the original project lead) wasn't very fond of GUIs 
when he wrote the original RA.  In 1.1 we are hoping to do a lot to 
change the "maziness", but it's still going to have a little bit of a 
learning curve.  The question is how to do that without working 
against the overall project goals.

If you're familiar with Wii hacking, you can kind of liken a libretro 
player somewhat to an IOS or a cIOS.  It's just a common layer used 
by the actual programs (your Wii .dol/.elf files can be compared with 
libretro cores…)  Console memory limitations notwithstanding, 
RetroArch is basically like the combination of an IOS, a launcher, 
and a menu.

The analogy isn't 1:1 because RA probably works more like the PS3 
does, and as noted on the Wii we have to do things to make RA look 
like one thing when it's not.  Although, it's worth noting that 
certain things on the Wii look like one cohesive thing and aren't.  
The Home menus for example.

My own contributions to this are taking the long view, which is why I 
have started with libretro-super as I have.  It might not all be done 
until 2.0 or 3.0, but when it gets there it should be as approachable 
as possible.  Far more so than Wii hacking itself is, IMO.

Joseph
